Sources/TuistCore/Cache/CacheDirectoriesProvider.swift

```diff
@@ -0,0 +1,64 @@
+import Foundation
+import TSCBasic
+import TuistGraph
+import TuistSupport
+
+public protocol CacheDirectoriesProviding: AnyObject {
+    /// Returns the cache directory
+    var cacheDirectory: AbsolutePath { get }
+
+    /// Returns the directory where the build artifacts are cached.
+    var buildCacheDirectory: AbsolutePath { get }
+
+    /// Returns the directory where hashes of modules that have been a part of successful test are cached
+    var testsCacheDirectory: AbsolutePath { get }
+    
+    /// Returns the directory where the projects generated for automation tasks are generated to
+    var projectsCacheDirectory: AbsolutePath { get }
+    
+    /// Returns the directory where the project description helper modules are cached.
+    var projectDescriptionHelpersCacheDirectory: AbsolutePath { get }
+    
+    /// Returns the directory where the project description helper modules are cached.
+    var manifestCacheDirectory: AbsolutePath { get }
+}
+
+public final class CacheDirectoriesProvider: CacheDirectoriesProviding {
+    public let cacheDirectory: AbsolutePath
+    private static let defaultDirectory = AbsolutePath(URL(fileURLWithPath: NSHomeDirectory()).path).appending(component: ".tuist")
+
+    public init(config: Config?) {
+        if let cacheDirectory = config?.cache?.path {
+            self.cacheDirectory = cacheDirectory
+        } else if let envVariable = ProcessInfo.processInfo.environment[Constants.EnvironmentVariables.cacheDirectory] {
```

Comment:
I think we can delete this variable. It was created for acceptance tests purposes but since we'll be able to define this via public API, it should no longer be necessary